vault/resource_consul_secret_backend_role_test.go

```diff
@@ -38,40 +45,155 @@ func TestConsulSecretBackendRole(t *testing.T) {
+					Steps: []resource.TestStep{
+						{
+							Config: testConsulSecretBackendRole_initialConfig(backendOld, nameOld, token, true, false),
+							Check:  resource.ComposeTestCheckFunc(createTestCheckFuncsOld...),
+						},
+						{
+							Config: testConsulSecretBackendRole_updateConfig(backendOld, nameOld, token, true, false),
+							Check:  resource.ComposeTestCheckFunc(updateTestCheckFuncsOld...),
+						},
+						{
+							ResourceName:      resourcePath,
+							ImportState:       true,
+							ImportStateVerify: true,
+						},
+					},
+				})
+			}
+		}
+	} else {
+		// If the VAULT_VERSION environment variable was not specified, assume they are using the latest version
+		versionTestFlag = true
+	}
+
+	if versionTestFlag {
+		resource.Test(t, resource.TestCase{
+			Providers:    testProviders,
+			PreCheck:     func() { testutil.TestAccPreCheck(t) },
+			CheckDestroy: testAccConsulSecretBackendRoleCheckDestroy,
+			Steps: []resource.TestStep{
+				{
+					Config:      testConsulSecretBackendRole_initialConfig(backend, name, token, false, false),
+					ExpectError: regexp.MustCompile(`Use either a policy document, a list of policies or roles, or a set of service or node identities, depending on your Consul version`),
+				},
+				{
+					Config:      testConsulSecretBackendRole_initialConfig(backend, name, token, true, true),
+					ExpectError: regexp.MustCompile(`Conflicting configuration arguments`),
+				},
+				{
+					Config: testConsulSecretBackendRole_initialConfig(backend, name, token, false, true),
+					Check:  resource.ComposeTestCheckFunc(createTestCheckFuncs...),
+				},
+				{
+					Config:      testConsulSecretBackendRole_updateConfig(backend, name, token, false, false),
+					ExpectError: regexp.MustCompile(`Use either a policy document, a list of policies or roles, or a set of service or node identities, depending on your Consul version`),
+				},
+				{
+					Config:      testConsulSecretBackendRole_updateConfig(backend, name, token, true, true),
+					ExpectError: regexp.MustCompile(`Conflicting configuration arguments`),
+				},
+				{
+					Config: testConsulSecretBackendRole_updateConfig(backend, name, token, false, true),
+					Check:  resource.ComposeTestCheckFunc(updateTestCheckFuncs...),
+				},
+				{
+					ResourceName:      resourcePath,
+					ImportState:       true,
+					ImportStateVerify: true,
+				},
 			},
-		},
-	})
+		})
+	}
+}
+
+func checkVaultVersionEnvPolicies() (bool, error) {
+	if val, ok := os.LookupEnv("VAULT_VERSION"); ok {
+		consulPoliciesVersion, _ := goversion.NewVersion("1.11")
+		envVersion, err := goversion.NewVersion(val)
+		if err != nil {
+			return true, fmt.Errorf("error parsing vault version from VAULT_VERSION environment variable: %v", err)
+		} else {
+			if envVersion.GreaterThanOrEqual(consulPoliciesVersion) {
+				return true, nil
+			}
+		}
+	} else {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func checkVaultVersionEnvConsulPolicies() (bool, error) {
```

Comment:
Isn't this the same as the `checkVaultVersionEnvPolicies` above? Can we consolidate some of the code into one function?

Reply:
Oh, I forgot to delete those functions. They are leftover code I forgot to remove I was first trying to use SkipFunc in the TestSteps. However, I couldn't find a way to implement SkipFunc with the tests in a way that worked because it needs to accept a parameter which breaks the expected function definition type.